From 01ef17e87dc7084d7fec44a07eb77f8d9e422f76 Mon Sep 17 00:00:00 2001
From: Bastian Blank <waldi@debian.org>
Date: Tue, 30 Jun 2015 11:46:08 +0200
Subject: force-paths

---
 configure.in | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/configure.in b/configure.in
index 4591891..37fae1c 100644
--- a/configure.in
+++ b/configure.in
@@ -1829,11 +1829,7 @@ if test "$BUILD_DMFILEMAPD" = yes; then
 fi
 
 ################################################################################
-AC_PATH_TOOL(MODPROBE_CMD, modprobe)
-
-if test -n "$MODPROBE_CMD"; then
-	AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["$MODPROBE_CMD"], [The path to 'modprobe', if available.])
-fi
+AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["/sbin/modprobe"], [The path to 'modprobe', if available.])
 
 
 lvm_exec_prefix=$exec_prefix
