From 06702fb9d015eb3d02a88582fc0a3afc48071d7c Mon Sep 17 00:00:00 2001
From: Bastian Blank <waldi@debian.org>
Date: Tue, 30 Jun 2015 11:46:08 +0200
Subject: force-paths

---
 configure.in | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/configure.in b/configure.in
index c7766b921..9864847d7 100644
--- a/configure.in
+++ b/configure.in
@@ -1821,11 +1821,7 @@ if test "$BUILD_DMFILEMAPD" = yes; then
 fi
 
 ################################################################################
-AC_PATH_TOOL(MODPROBE_CMD, modprobe)
-
-if test -n "$MODPROBE_CMD"; then
-	AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["$MODPROBE_CMD"], [The path to 'modprobe', if available.])
-fi
+AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["/sbin/modprobe"], [The path to 'modprobe', if available.])
 
 
 lvm_exec_prefix=$exec_prefix
