From e63e6bbdb68aa3cc9143aed086ae79e731856089 Mon Sep 17 00:00:00 2001
From: Bastian Blank <waldi@debian.org>
Date: Tue, 30 Jun 2015 12:43:46 +0200
Subject: cleanup

---
 make.tmpl.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/make.tmpl.in b/make.tmpl.in
index ce0d599..b75a6a1 100644
--- a/make.tmpl.in
+++ b/make.tmpl.in
@@ -482,7 +482,7 @@ else
 	for i in $(EXPORTED_SYMBOLS); do\
 		echo "$${i##*.} {"; echo "	global:";\
 		$(SED) "s/^/		/;s/$$/;/" $$i;\
-		test "$$i" = Base && { echo "	local:"; echo "		*;"; };\
+		test "$${i##*.}" = Base && { echo "	local:"; echo "		*;"; };\
 		echo "};";\
 	done > $@
 endif
