From 6d5f004ddcb47967bc7d4d8b0ed20b570f07ad90 Mon Sep 17 00:00:00 2001
From: Marian Csontos <mcsontos@redhat.com>
Date: Thu, 6 Dec 2018 10:50:42 +0100
Subject: udev: 69-dm-lvm-metad.rules is still needed

---
 udev/Makefile.in | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/udev/Makefile.in b/udev/Makefile.in
index 3a14fb5d89..2d25174c32 100644
--- a/udev/Makefile.in
+++ b/udev/Makefile.in
@@ -18,10 +18,7 @@ top_builddir = @top_builddir@
 include $(top_builddir)/make.tmpl
 
 DM_RULES=55-dm.rules 60-persistent-storage-dm.rules 95-dm-notify.rules
-LVM_RULES=56-lvm.rules
-ifeq ("@BUILD_LVMETAD@", "yes")
-LVM_RULES+=69-lvm-metad.rules
-endif
+LVM_RULES=56-lvm.rules 69-lvm-metad.rules
 
 DM_DIR=$(shell $(GREP) "\#define DM_DIR" $(top_srcdir)/libdm/misc/dm-ioctl.h | $(AWK) '{print $$3}')
 
