Index: debian/configure.in
===================================================================
--- debian.orig/configure.in	2008-07-01 16:52:35.000000000 +0100
+++ debian/configure.in	2008-07-01 16:52:45.000000000 +0100
@@ -519,11 +519,7 @@
 AC_CHECK_HEADERS(libdevmapper.h,,AC_MSG_ERROR(bailing out))
 
 ################################################################################
-AC_PATH_PROG(MODPROBE_CMD, modprobe)
-
-if test x$MODPROBE_CMD != x; then
-	AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["$MODPROBE_CMD"], [The path to 'modprobe', if available.])
-fi
+AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["/sbin/modprobe"], [The path to 'modprobe', if available.])
 
 ################################################################################
 LVM_VERSION="\"`cat VERSION 2>/dev/null || echo Unknown`\""
