--- a/configure.in
+++ b/configure.in
@@ -413,45 +413,13 @@
   *) AC_MSG_ERROR([--with-thin parameter invalid ($THIN)]) ;;
 esac
 
-case "$THIN" in
-  internal|shared)
-	AC_ARG_WITH(thin-check,
-		AC_HELP_STRING([--with-thin-check=PATH],
-			       [thin_check tool: [[autodetect]]]),
-			       THIN_CHECK_CMD=$withval, THIN_CHECK_CMD="autodetect")
-	# Empty means a config way to ignore thin checking
-	if test "$THIN_CHECK_CMD" = "autodetect"; then
-		AC_PATH_PROG(THIN_CHECK_CMD, thin_check)
-		test -z "$THIN_CHECK_CMD" && AC_MSG_ERROR(thin_check not found in path $PATH)
-	fi
-	AC_ARG_WITH(thin-dump,
-		AC_HELP_STRING([--with-thin-dump=PATH],
-			       [thin_dump tool: [[autodetect]]]),
-			       THIN_DUMP_CMD=$withval, THIN_DUMP_CMD="autodetect")
-	# Empty means a config way to ignore thin checking
-	if test "$THIN_DUMP_CMD" = "autodetect"; then
-		AC_PATH_PROG(THIN_DUMP_CMD, thin_dump)
-		test -z "$THIN_DUMP_CMD" && AC_MSG_ERROR(thin_dump not found in path $PATH)
-	fi
-	AC_ARG_WITH(thin-repair,
-		AC_HELP_STRING([--with-thin-repair=PATH],
-			       [thin_repair tool: [[autodetect]]]),
-			       THIN_REPAIR_CMD=$withval, THIN_REPAIR_CMD="autodetect")
-	# Empty means a config way to ignore thin checking
-	if test "$THIN_REPAIR_CMD" = "autodetect"; then
-		AC_PATH_PROG(THIN_REPAIR_CMD, thin_repair)
-		test -z "$THIN_REPAIR_CMD" && AC_MSG_ERROR(thin_repair not found in path $PATH)
-	fi
-	;;
-esac
-
-AC_DEFINE_UNQUOTED([THIN_CHECK_CMD], ["$THIN_CHECK_CMD"],
+AC_DEFINE_UNQUOTED([THIN_CHECK_CMD], ["/usr/sbin/thin_check"],
 		   [The path to 'thin_check', if available.])
 
-AC_DEFINE_UNQUOTED([THIN_DUMP_CMD], ["$THIN_DUMP_CMD"],
+AC_DEFINE_UNQUOTED([THIN_DUMP_CMD], ["/usr/sbin/thin_dump"],
 		   [The path to 'thin_dump', if available.])
 
-AC_DEFINE_UNQUOTED([THIN_REPAIR_CMD], ["$THIN_REPAIR_CMD"],
+AC_DEFINE_UNQUOTED([THIN_REPAIR_CMD], ["/usr/sbin/thin_repair"],
 		   [The path to 'thin_repair', if available.])
 
 
@@ -1341,11 +1309,7 @@
 fi
 
 ################################################################################
-AC_PATH_PROG(MODPROBE_CMD, modprobe)
-
-if test x$MODPROBE_CMD != x; then
-	AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["$MODPROBE_CMD"], [The path to 'modprobe', if available.])
-fi
+AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["/sbin/modprobe"], [The path to 'modprobe', if available.])
 
 
 lvm_exec_prefix=$exec_prefix
