Index: debian/tools/Makefile.in
===================================================================
--- debian.orig/tools/Makefile.in	2008-07-01 16:53:18.000000000 +0100
+++ debian/tools/Makefile.in	2008-07-01 16:53:23.000000000 +0100
@@ -141,22 +141,16 @@
 endif
 
 install_cmdlib_dynamic: liblvm2cmd.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) liblvm2cmd.so \
-		$(libdir)/liblvm2cmd.so.$(LIB_VERSION)
+	$(INSTALL) -D liblvm2cmd.so $(libdir)/liblvm2cmd.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2cmd.so.$(LIB_VERSION) $(libdir)/liblvm2cmd.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 444 lvm2cmd.h \
-		$(includedir)/lvm2cmd.h
+	$(INSTALL) -D -m 644 lvm2cmd.h $(includedir)/lvm2cmd.h
 
 install_cmdlib_static: liblvm2cmd-static.a
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) liblvm2cmd-static.a \
-		$(libdir)/liblvm2cmd.a.$(LIB_VERSION)
-	$(LN_S) -f liblvm2cmd.a.$(LIB_VERSION) $(libdir)/liblvm2cmd.a
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 444 lvm2cmd.h \
-		$(includedir)/lvm2cmd.h
+	$(INSTALL) -D -m 644 liblvm2cmd-static.a $(libdir)/liblvm2cmd.a
+	$(INSTALL) -D -m 644 lvm2cmd.h $(includedir)/lvm2cmd.h
 
 install_tools_dynamic: lvm .commands
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) lvm \
-		$(sbindir)/lvm
+	$(INSTALL) -D lvm $(sbindir)/lvm
 	@echo Creating symbolic links for individual commands in $(sbindir)
 	@( \
 		for v in `cat .commands`; do \
@@ -166,7 +160,6 @@
 	)
 
 install_tools_static: lvm.static
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) lvm.static \
-		$(staticdir)/lvm.static
+	$(INSTALL) -D lvm.static $(staticdir)/lvm.static
 
 install: $(INSTALL_TARGETS)
Index: debian/daemons/clvmd/Makefile.in
===================================================================
--- debian.orig/daemons/clvmd/Makefile.in	2008-07-01 16:53:18.000000000 +0100
+++ debian/daemons/clvmd/Makefile.in	2008-07-01 16:53:23.000000000 +0100
@@ -89,8 +89,7 @@
 .PHONY: install_clvmd
 
 install_clvmd: $(TARGETS)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) clvmd \
-		$(sbindir)/clvmd
+	$(INSTALL) -D clvmd $(sbindir)/clvmd
 
 install: $(INSTALL_TARGETS)
 
Index: debian/dmeventd/mirror/Makefile.in
===================================================================
--- debian.orig/dmeventd/mirror/Makefile.in	2008-07-01 16:53:18.000000000 +0100
+++ debian/dmeventd/mirror/Makefile.in	2008-07-01 16:53:23.000000000 +0100
@@ -30,6 +30,5 @@
 include $(top_srcdir)/make.tmpl
 
 install: libdevmapper-event-lvm2mirror.$(LIB_SUFFIX)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
-		$(libdir)/$<.$(LIB_VERSION)
+	$(INSTALL) -D $< $(libdir)/$<.$(LIB_VERSION)
 	$(LN_S) -f $<.$(LIB_VERSION) $(libdir)/$<
Index: debian/lib/format1/Makefile.in
===================================================================
--- debian.orig/lib/format1/Makefile.in	2008-07-01 16:53:18.000000000 +0100
+++ debian/lib/format1/Makefile.in	2008-07-01 16:53:23.000000000 +0100
@@ -30,7 +30,6 @@
 include $(top_srcdir)/make.tmpl
 
 install: liblvm2format1.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
-		$(libdir)/liblvm2format1.so.$(LIB_VERSION)
+	$(INSTALL) -D $< $(libdir)/liblvm2format1.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2format1.so.$(LIB_VERSION) $(libdir)/liblvm2format1.so
 
Index: debian/lib/format_pool/Makefile.in
===================================================================
--- debian.orig/lib/format_pool/Makefile.in	2008-07-01 16:53:18.000000000 +0100
+++ debian/lib/format_pool/Makefile.in	2008-07-01 16:53:23.000000000 +0100
@@ -27,8 +27,7 @@
 include $(top_srcdir)/make.tmpl
 
 install: liblvm2formatpool.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
-		$(libdir)/liblvm2formatpool.so.$(LIB_VERSION)
+	$(INSTALL) -D $< $(libdir)/liblvm2formatpool.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2formatpool.so.$(LIB_VERSION) \
 		$(libdir)/liblvm2formatpool.so
 
Index: debian/lib/locking/Makefile.in
===================================================================
--- debian.orig/lib/locking/Makefile.in	2008-07-01 16:53:18.000000000 +0100
+++ debian/lib/locking/Makefile.in	2008-07-01 16:53:23.000000000 +0100
@@ -23,8 +23,7 @@
 include $(top_srcdir)/make.tmpl
 
 install install_cluster: liblvm2clusterlock.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
-		$(libdir)/liblvm2clusterlock.so.$(LIB_VERSION)
+	$(INSTALL) -D $< $(libdir)/liblvm2clusterlock.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2clusterlock.so.$(LIB_VERSION) \
 		$(libdir)/liblvm2clusterlock.so
 
Index: debian/lib/mirror/Makefile.in
===================================================================
--- debian.orig/lib/mirror/Makefile.in	2008-07-01 16:53:18.000000000 +0100
+++ debian/lib/mirror/Makefile.in	2008-07-01 16:53:23.000000000 +0100
@@ -23,7 +23,6 @@
 include $(top_srcdir)/make.tmpl
 
 install: liblvm2mirror.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
-		$(libdir)/liblvm2mirror.so.$(LIB_VERSION)
+	$(INSTALL) -D $< $(libdir)/liblvm2mirror.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2mirror.so.$(LIB_VERSION) $(libdir)/liblvm2mirror.so
 
Index: debian/lib/snapshot/Makefile.in
===================================================================
--- debian.orig/lib/snapshot/Makefile.in	2008-07-01 16:53:18.000000000 +0100
+++ debian/lib/snapshot/Makefile.in	2008-07-01 16:53:23.000000000 +0100
@@ -23,8 +23,7 @@
 include $(top_srcdir)/make.tmpl
 
 install: liblvm2snapshot.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
-		$(libdir)/liblvm2snapshot.so.$(LIB_VERSION)
+	$(INSTALL) -D $< $(libdir)/liblvm2snapshot.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2snapshot.so.$(LIB_VERSION) \
 		$(libdir)/liblvm2snapshot.so
 
Index: debian/man/Makefile.in
===================================================================
--- debian.orig/man/Makefile.in	2008-07-01 16:53:18.000000000 +0100
+++ debian/man/Makefile.in	2008-07-01 16:53:23.000000000 +0100
@@ -46,14 +46,14 @@
 	@for f in $(MAN8); \
 	do \
 	   $(RM) $(MAN8DIR)/$$f; \
-	   @INSTALL@ -D $(OWNER) $(GROUP) -m 444 $$f $(MAN8DIR)/$$f; \
+	   @INSTALL@ -D $(OWNER) $(GROUP) -m 644 $$f $(MAN8DIR)/$$f; \
 	done
 
 	@echo "Installing $(MAN5) in $(MAN5DIR)"
 	@for f in $(MAN5); \
 	do \
 	   $(RM) $(MAN5DIR)/$$f; \
-	   @INSTALL@ -D $(OWNER) $(GROUP) -m 444 $$f $(MAN5DIR)/$$f; \
+	   @INSTALL@ -D $(OWNER) $(GROUP) -m 644 $$f $(MAN5DIR)/$$f; \
 	done
 
 install_cluster:
@@ -61,5 +61,5 @@
 	@for f in $(MAN8CLUSTER); \
 	do \
 	   $(RM) $(MAN8DIR)/$$f; \
-	   @INSTALL@ -D $(OWNER) $(GROUP) -m 444 $$f $(MAN8DIR)/$$f; \
+	   @INSTALL@ -D $(OWNER) $(GROUP) -m 644 $$f $(MAN8DIR)/$$f; \
 	done
