Description: add micron PCIe filter
Author: Dmitrijs Ledkovs <xnox@ubuntu.com
Origin: http://git.fedorahosted.org/cgit/lvm2.git/patch/?id=7803756e9751117c8a2454d55c145f6e759279dd
Bug-Ubuntu: https://launchpad.net/bugs/1122445

--- lvm2-2.02.95.orig/lib/filters/filter.c
+++ lvm2-2.02.95/lib/filters/filter.c
@@ -136,6 +136,7 @@ static const device_info_t device_info[]
 	{"mmc", 16},		/* MMC block device */
 	{"blkext", 1},		/* Extended device partitions */
 	{"fio", 16},		/* Fusion */
+	{"mtip32xx", 16},	/* Micron PCIe SSDs */
 	{NULL, 0}
 };
 
